packages/frontend/amp/lib/ad-json.ts

```diff
@@ -8,22 +8,17 @@ interface AdJson {
     targeting: KV[];
 }
 
-export const adJson = (
-    edition: Edition,
-    editionAdTargetings: EditionAdTargeting[],
-): AdJson => {
-    const targeting = editionAdTargetings.find(t => t.edition === edition);
-
+export const adJson = (targeting: AdTargetParam[]): AdJson => {
     if (!targeting) {
         return { targeting: [] };
     }
 
-    let json = targeting.paramSet.map(p => ({
-        name: p.name,
-        value: p.values.join(','),
-    }));
-
-    json = json.filter(p => p.name !== 'p');
+    const json = targeting
+        .map(({ name, value }) => ({
+            name,
+            value: Array.isArray(value) ? value.join(',') : value,
+        }))
+        .filter(p => p.name !== 'p');
```

Comment:
Would it be marginally more efficient to filter out the items where the `name` isn't equal to "p" before modifying the `value` properties in the `map`? Something like...

```js
const json = targeting
.filter(p => p.name !== 'p')
.map(({ name, value }) => ({
            name,
            value: Array.isArray(value) ? value.join(',') : value,
        }))';
```